framework/src/main/java/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -549,12 +578,14 @@ private AnnotationMirror createAnnotationForStringConcatenation(
         if (leftValues != null && rightValues != null) {
             // Both operands have known string values, compute set of results
             List<String> concatValues = new ArrayList<>();
-            if (leftValues.isEmpty()) {
-                leftValues = Collections.singletonList("null");
+
+            if (!isNotNullable(leftOperand)) {
```

Comment:
That every time this is called there is a `!` first suggests that the method should be called "isNullable"